### PR TITLE
Keycloak

### PR DESCRIPTION
**Pull request by SaiDadireddy**
## Description of the change

This PR adds `Keycloak` as an identity provider type. Adds interfaces needed for groups integrations.

## Relevant release note information

Release Notes: Keycloak as a supported IdP

## Related JIRA tickets

Relates to JIRA: CWC-2517

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [X] No

If yes, please explain:

### CODE REVIEW

**Review by tflpd — APPROVED** [review]

File: mrtap.service/mrtap.service.types.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export interface MrtapConfigSchema {
     orgProvider: string
 }
 
-export function getDefaultMrtapConfig(): MrtapConfigSchema {
+export async function getDefaultMrtapConfig(): Promise<MrtapConfigSchema> {
```
Comment (lipmas) [inline mrtap.service/mrtap.service.types.ts:11]: What is the reason for this change?

Comment (SaiDadireddy) [reply mrtap.service/mrtap.service.types.ts:11]: The `getMrtap` function was updated to return a promise [here](https://github.com/bastionzero/webshell-common-ts/pull/112). There wasn't a PR to update this function to return a promise in our backend so it was causing conflicts for me. I fixed that [here](https://github.com/bastionzero/webshell-backend/pull/1573/commits/957c950bfb19073a049e827d2e2908032f039f24) in the backend, and these webshell changes were needed for that.

Comment (SaiDadireddy) [reply mrtap.service/mrtap.service.types.ts]: Ok I resolved it by making the getmrtap async [21336c4](https://github.com/bastionzero/webshell-backend/pull/1573/commits/21336c48759c5809708781f1c39da3c441d28e91) the backend and removed async from this function [319426d](https://github.com/bastionzero/webshell-common-ts/pull/113/commits/319426df16ea305a5f820c9c0fb1b85130e08cd3)

**Review by lgmugnier — APPROVED** [review]

**Review by lipmas — APPROVED** [review]
